WalletWasabi/Services/Terminate/TerminateService.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Services.Terminate
+{
+	public class TerminateService
+	{
+		private Func<Task> _terminateApplicationAsync;
+
+		private const long TerminateStatusIdle = 0;
```

Comment:
Idle can be understood wrongly here as "not doing anything" but in any phase of the termination process. I would say it's better to use a name that suggests that "the termination process has not been started yet".

Reply:
Yes, better.